lib/resourcemerge/core.go

```diff
@@ -734,12 +726,10 @@ func setBool(modified *bool, existing *bool, required bool) {
 }
 
 func setBoolPtr(modified *bool, existing **bool, required *bool) {
-	// if we have no required, then we don't care what someone else has set
-	if required == nil {
+	if *existing == nil && required == nil {
```

Comment:
This change will also tighten a bunch of other places:

```console
$ git grep '^func \|setBoolPtr' origin/pr/804 | grep -B1 setBoolPtr
origin/pr/804:lib/resourcemerge/batch.go:func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
origin/pr/804:lib/resourcemerge/batch.go:       setBoolPtr(modified, &existing.Spec.ManualSelector, required.Spec.ManualSelector)
--
origin/pr/804:lib/resourcemerge/core.go:func EnsureServiceAccount(modified *bool, existing *corev1.ServiceAccount, required corev1.ServiceAccount) {
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.AutomountServiceAccountToken, required.AutomountServiceAccountToken)
--
origin/pr/804:lib/resourcemerge/core.go:func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.PodSpec) {
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.ShareProcessNamespace, required.ShareProcessNamespace)
--
origin/pr/804:lib/resourcemerge/core.go:func ensureSecurityContext(modified *bool, existing *corev1.SecurityContext, required corev1.SecurityContext) {
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.Privileged, required.Privileged)
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.RunAsNonRoot, required.RunAsNonRoot)
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.ReadOnlyRootFilesystem, required.ReadOnlyRootFilesystem)
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.AllowPrivilegeEscalation, required.AllowPrivilegeEscalation)
--
origin/pr/804:lib/resourcemerge/core.go:func ensurePodSecurityContext(modified *bool, existing *corev1.PodSecurityContext, required corev1.PodSecurityContext) {
origin/pr/804:lib/resourcemerge/core.go:        setBoolPtr(modified, &existing.RunAsNonRoot, required.RunAsNonRoot)
--
origin/pr/804:lib/resourcemerge/core.go:func setBool(modified *bool, existing *bool, required bool) {
origin/pr/804:lib/resourcemerge/core.go:func setBoolPtr(modified *bool, existing **bool, required *bool) {
--
origin/pr/804:lib/resourcemerge/security.go:func EnsureSecurityContextConstraints(modified *bool, existing *securityv1.SecurityContextConstraints, required securityv1.SecurityContextConstraints) {
origin/pr/804:lib/resourcemerge/security.go:    setBoolPtr(modified, &existing.DefaultAllowPrivilegeEscalation, required.DefaultAllowPrivilegeEscalation)
origin/pr/804:lib/resourcemerge/security.go:    setBoolPtr(modified, &existing.AllowPrivilegeEscalation, required.AllowPrivilegeEscalation)
```

That's probably fine, but it may be worth calling out Job's `spec.manualSelector` and similar in the commit message.

Reply:
Right, I should've mention `Kind`s `Job` and `DaemonSet`

Reply:
changed